Show tech badges on "work" page

artemave opened the pull request:
So that people can see at a glance what tech we've been using in the past.

artemave committed:
first ribbon: iOS

artemave committed:
The rest of the badges

artemave committed:
Make all dependenceis production

So we can run this on heroku.

artemave committed:
Tweek scripts so it can run on heroku

artemave committed:
ribbons -> badges

AlecRust commented on package.json:
```diff
@@ -4,7 +4,7 @@
   "description": "Static Featurist Ltd site.",
   "scripts": {
-    "postinstall": "yarn build",
+    "heroku-postbuild": "yarn svg && yarn css && yarn js",
```
Have you moved it to Heroku or something?

artemave replied:
Nope, heroku is just an easy way to show off a branch.

AlecRust commented on package.json:
```diff
@@ -18,20 +18,12 @@
     "deploy": "yarn build-production && gh-pages -d dist -b gh-pages"
   },
   "repository": "featurist:featurist-site",
   "dependencies": {
```
No more devDependencies?

artemave replied:
Heroku deploy needs to build assets and so rather than cherry picking some packages from devDependencies into dependencies I thought I'd just move them all because it does not affect production anyway.

AlecRust commented on src/images/_svg/angular.svg:
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
I would pass these through [SVGOMG](https://jakearchibald.github.io/svgomg/), all these SVGs will be in the HTML of every page.

artemave replied:
I am not following! Do you mind adding that change?

AlecRust replied:
Sure done in 67eff0d.

AlecRust commented on src/_css/components/badge.css:
```diff
@@ -0,0 +1,20 @@
+  margin-left: 10px;
+  width: 44px;
+
+  svg {
```
We have an `icon.css` component - do we need to target the `svg` element directly here?

artemave replied:
Ah, well, no idea... Do you mind changing it? :)

AlecRust replied:
Sure done in f608e54.

artemave committed:
Add grey bg to the tech badge container

Also, center align badges.

artemave committed:
Fix incosistence media-hero height

AlecRust committed:
Optimise new SVG assets with SVGOMG

https://jakearchibald.github.io/svgomg/

AlecRust committed:
Refactor badges to use .Icon

- Refactor naming in badges component
- Refactor `.MediaHero`/`.MediaHeroList` styling to support `.Badges` as sibling of `.MediaHero`.

AlecRust commented on src/work/index.ejs:
```diff
@@ -14,7 +14,7 @@
       <li class="MediaHeroList-item">
         <figure class="MediaHero">
-          <a href="<%= slug %>/">
+          <a href="<%= slug %>">
```
Slash was purposeful here. This will cause a (minor) redirect.

artemave replied:
Ah... Sorry! :)

AlecRust replied:
Made this more obvious in 8620ca0.

deioo committed:
Tweak the badges background a tiny bit

artemave commented:
For posterity: svgs are taken from here: https://svgporn.com/

deioo commented on src/_css/components/badges.css:
```diff
@@ -0,0 +1,20 @@
+
+.Badges {
+  background-color: #fafafa;
+  box-shadow: inset 0 10px 20px -10px rgba(10, 10, 10, 0.075);
```
This produces this beaut

<img width="1242" alt="screenshot 2018-12-11 at 17 09 49" src="https://user-images.githubusercontent.com/1321353/49817892-5983bf80-fd69-11e8-80ca-01f15520e633.png">

artemave commented:
@AlecRust when we add more badges, how should we "optimize" them?

AlecRust commented:
@artemave you can paste the original in to [SVGOMG](https://jakearchibald.github.io/svgomg/) and use the output in our repo. It's just the equivalent of binary image compression.

We do actually optimise SVGs in production already, this is just a minor consistency thing having uniform source SVGs.

AlecRust committed:
Refactor how slash is added to end of URL

artemave committed:
Few more badges
